src/dart/parse.rs

```diff
@@ -200,14 +221,29 @@ impl<'a> Parser<'a> {
         Ok(())
     }
 
+    /// Returns true and consumes the current token
+    /// if it matches the given keyword.
     pub fn eat_keyword(&mut self, s: &'static str) -> bool {
         self.expect_keyword(s).is_ok()
     }
 
+    /// Returns true if the current token matches the given keyword.
+    pub fn is_keyword(&self, s: &str) -> bool {
+        if let Some(token) = self.cur {
+            token.as_ident().map_or(false, |x| x == s)
+        } else {
+            false
+        }
+    }
+
+    /// Speculatively executes a closure and passes a parser to it
+    /// and preserves parser state.
     pub fn probe<F: FnOnce(&mut Self) -> R, R>(&self, f: F) -> R {
         f(&mut self.clone())
     }
 
+    /// Executes a closure that commits the modified state on a success,
```

Comment:
Still need to figure out how to refer to the closure here, maybe by avoiding "closure":
"Applies `f` on the parser, committing the modified state on success, and rolling back on error."